keras/utils/timeseries_dataset.py

```diff
@@ -212,47 +214,29 @@ def timeseries_dataset_from_array(
     num_seqs = end_index - start_index - (sequence_length * sampling_rate) + 1
     if targets is not None:
         num_seqs = min(num_seqs, len(targets))
-    if num_seqs < 2147483647:
-        index_dtype = "int32"
-    else:
-        index_dtype = "int64"
-
-    # Generate start positions
-    start_positions = np.arange(0, num_seqs, sequence_stride, dtype=index_dtype)
-    if shuffle:
-        if seed is None:
-            seed = np.random.randint(1e6)
-        rng = np.random.RandomState(seed)
-        rng.shuffle(start_positions)
-
-    sequence_length = tf.cast(sequence_length, dtype=index_dtype)
-    sampling_rate = tf.cast(sampling_rate, dtype=index_dtype)
 
-    positions_ds = tf.data.Dataset.from_tensors(start_positions).repeat()
-
-    # For each initial window position, generates indices of the window elements
-    indices = tf.data.Dataset.zip(
-        (tf.data.Dataset.range(len(start_positions)), positions_ds)
-    ).map(
-        lambda i, positions: tf.range(
-            positions[i],
-            positions[i] + sequence_length * sampling_rate,
-            sampling_rate,
-        ),
-        num_parallel_calls=tf.data.AUTOTUNE,
+    dataset = tf.data.Dataset.from_tensor_slices(data[start_index:end_index])
+    dataset = dataset.window(
```

Comment:
This is a big departure from the previous implementation. Could you provide a quick performance benchmark of each version on a large dataset of long timeseries?

Reply:
I don't have enough computation power and large dataset. Please can you provide it by yourself? Thanks a lot for your time.

Reply:
Anyone from the community is welcome to continue this work to do the benchmarks. We do not have immediate bandwidth to do the benchmarking.